x-pack/plugins/search_assistant/server/plugin.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+} from '@kbn/core/server';
+
+import type { SearchAssistantPluginSetup, SearchAssistantPluginStart } from './types';
+import { defineRoutes } from './routes';
+
+export class SearchAssistantPlugin
+  implements Plugin<SearchAssistantPluginSetup, SearchAssistantPluginStart>
+{
+  private readonly logger: Logger;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+  }
+
+  public setup(core: CoreSetup) {
+    this.logger.debug('SearchAssistant: Setup');
```

Comment:
Let's remove this logging, that's just clutter in the console that doesn't add anything. While we're at it, we should remove the next lines about creating a router and defining routes as well. It's too soon for that: let's just create the plugin and nothing else. Keep it small.